x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/components/endpoint_details_tabs.tsx

```diff
@@ -47,15 +51,42 @@ const StyledEuiTabbedContent = styled(EuiTabbedContent)`
 
 export const EndpointDetailsFlyoutTabs = memo(
   ({ show, tabs }: { show: EndpointIndexUIQueryParams['show']; tabs: EndpointDetailsTabs[] }) => {
+    const dispatch = useDispatch<(action: EndpointAction) => void>();
+    const { pageSize } = useEndpointSelector(getActivityLogDataPaging);
     const [selectedTabId, setSelectedTabId] = useState<EndpointDetailsTabsId>(() => {
       return show === 'details'
         ? EndpointDetailsTabsTypes.overview
         : EndpointDetailsTabsTypes.activityLog;
     });
 
     const handleTabClick = useCallback(
-      (tab: EuiTabbedContentTab) => setSelectedTabId(tab.id as EndpointDetailsTabsId),
-      [setSelectedTabId]
+      (tab: EuiTabbedContentTab) => {
+        dispatch({
+          type: 'endpointDetailsFlyoutTabChanged',
+          payload: {
+            flyoutView: tab.id as EndpointIndexUIQueryParams['show'],
+          },
+        });
+        if (tab.id === EndpointDetailsTabsTypes.activityLog) {
+          const paging = {
+            page: 1,
+            pageSize,
+          };
+          dispatch({
```

Comment:
Why do you need to do two dispatch? Could this be done with just one?

Reply:
We can, but I wanted to be explicit about when the action triggers just a paging update vs when it actually triggers a fetch request.

Reply:
Ok, so page update doesn't fetch the request right? Yo have to do the fetch explicitly right?

Reply:
Yeah `endpointDetailsActivityLogUpdatePaging` just toggles the paging ability via a flag `disabled`. The other one (`appRequestedEndpointActivityLog`) updates paging numbers which is picked up in the middleware. Essentially the action names are separate for clarity of purpose. They can be merged into one and then do a relevant conditional check in the middleware like you suggest presumably. I prefer separate action names though.